client/services/FeedbackGeneratorService.js

```diff
@@ -53,94 +55,102 @@ tie.factory('FeedbackGeneratorService', [
+            var allowedOutputExample = (
+              correctnessTests[i].getAnyAllowedOutput());
+            var feedback = FeedbackObjectFactory.create(false);
+            feedback.appendTextParagraph([
+              'Your code gave the output ',
+              _jsToHumanReadable(observedOutput),
+              ' for the input ',
+              _jsToHumanReadable(correctnessTests[i].getInput()),
+              ' ... but this does not match the expected output ',
+              _jsToHumanReadable(allowedOutputExample),
+              '.'
+            ].join(''));
+            return feedback;
           }
+        }
 
-          var performanceTests = prompt.getPerformanceTests();
-          var performanceTestResults =
-              codeEvalResult.getPerformanceTestResults();
-          for (var i = 0; i < performanceTests.length; i++) {
-            var expectedPerformance = performanceTests[i].getExpectedPerformance();
-            var observedPerformance = performanceTestResults[i];
+        var performanceTests = prompt.getPerformanceTests();
+        var performanceTestResults =
+            codeEvalResult.getPerformanceTestResults();
+        for (var i = 0; i < performanceTests.length; i++) {
+          var expectedPerformance = performanceTests[i].getExpectedPerformance();
+          var observedPerformance = performanceTestResults[i];
 
-            if (expectedPerformance !== observedPerformance) {
-              var feedback = FeedbackObjectFactory.create(false);
-              feedback.appendTextParagraph([
-                'Your code is running more slowly than expected. Can you ',
-                'reconfigure it such that it runs in ',
-                expectedPerformance,
-                ' time?'
-              ].join(''));
-              return feedback;
-            }
+          if (expectedPerformance !== observedPerformance) {
+            var feedback = FeedbackObjectFactory.create(false);
+            feedback.appendTextParagraph([
+              'Your code is running more slowly than expected. Can you ',
+              'reconfigure it such that it runs in ',
+              expectedPerformance,
+              ' time?'
+            ].join(''));
+            return feedback;
           }
-
-          var feedback = FeedbackObjectFactory.create(true);
-          feedback.appendTextParagraph([
-            'You\'ve completed all the tasks for this question! Click the ',
-            '"Next" button to move on to the next question.',
-          ].join(''));
-          return feedback;
         }
+
+        var feedback = FeedbackObjectFactory.create(true);
+        feedback.appendTextParagraph([
+          'You\'ve completed all the tasks for this question! Click the ',
+          '"Next" button to move on to the next question.',
+        ].join(''));
+        return feedback;
+      }
+    };
+
+    return {
+      _prepareFeedback: _prepareFeedback,
+      getFeedback: function(prompt, codeEvalResult) {
```

Comment:
Let's do the transcript stuff in SolutionHandlerService -- that's the service that just does a bunch of "general management" of things and pushes stuff into various pipelines. I think that's the right "seam".